src/Parsers/GetObject/Multiple/Streamed.php

```diff
@@ -0,0 +1,93 @@
+<?php namespace PHRETS\Parsers\GetObject\Multiple;
+
+use GuzzleHttp\Psr7\Response;
+use PHRETS\Http\Response as PHRETSResponse;
+use Illuminate\Support\Collection;
+use PHRETS\Parsers\GetObject\Single;
+
+class Streamed
+{
+    // in bytes
+    // TODO: parameterize this
+    private const CHUNK_SIZE = 20 * 1000 * 1000;
+
+    public function parse(PHRETSResponse $response) : \Generator
+    {
+        if (!$response->getBody()) {
+            return;
+        }
+
+        $boundary = $this->determineBoundary($response->getHeader('Content-Type'));
+        $singleObjectParser = new Single();
+
+        $firstChunk = true;
+        $inProgressSegment = '';
+
+        while (!$response->getBody()->eof()) {
+            $chunk = $response->getBody()->read(self::CHUNK_SIZE);
+
+            // in all likelihood our previous chunk split a segment in half
+            // this re-combines that segment from its two halves
+            // NOTE: a single gigantic image (>100MB) could be problematic.
+            // returning gigantic images as streams would require a drastically different API
+            // e.g. first sending the image headers, then wrapping the HTTP stream
+            // with our own stream (in order to stop it at the boundary), or
+            // streaming the images to disk and providing userspace with filenames
+            $chunk = $inProgressSegment . $chunk;
+
+            $segments = preg_split("/(\r\n|^)--{$boundary}-?-?(\r\n|$)/", $chunk);
```

Comment:
> I don't know in what situations...

Looks like it's on invalid regex, which is within the realm of possibility here since the MLS specifies part of the boundary, good call